packages/plugin-ext/src/main/node/paths/plugin-paths-service.ts

```diff
@@ -95,10 +95,24 @@ export class PluginPathsServiceImpl implements PluginPathsService {
                 displayName = displayName.slice(0, displayName.lastIndexOf('.'));
             }
 
-            return crypto.createHash('md5').update(workspaceUri).digest('hex');
+            return this.createHash(workspaceUri);
         }
     }
 
+    /**
+     * Creates a hash digest of the given string.
+     */
+    protected createHash(str: string): string {
+        try {
+            // md5 is not FIPS-approved but we have to continue use it as there're existing storage folders based on it
+            return crypto.createHash('md5').update(str).digest('hex');
+        } catch (e) {
+            // FIPS-compliant
```

Comment:
Can we narrow down the error and log in all other cases? to avoid false positively creating sha256 hashes?

Reply:
done